File: backend/models/product.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Product extends Model {
     
    static associate(models) {
      // define association here
      Product.belongsTo(models.Category, {
        foreignKey: 'categoryId', as: 'category'
      })
      Product.belongsTo(models.User, {
        foreignKey: 'createdBy', as: 'Creator'
      })
      Product.belongsTo(models.User, {
        foreignKey: 'updatedBy', as: 'Updater'
      })
    }
  }
  Product.init({
    name: DataTypes.STRING,
    qty: DataTypes.INTEGER,
    categoryId: DataTypes.INTEGER,
    urlImage: DataTypes.STRING,
    createdBy: DataTypes.STRING,
    updatedBy: DataTypes.STRING
  }, {
    sequelize,
    modelName: 'Product',
  });
  return Product;
};

File: frontend/storage_management_system/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final int qty;
  final int categoryId;
  final String? urlImage;

  Product({
    required this.id,
    required this.name,
    required this.qty,
    required this.categoryId,
    this.urlImage,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      qty: json['qty'],
      categoryId: json['categoryId'],
      urlImage: json['urlImage'],
    );
  }

  get updatedBy => null;

  get createdBy => null;

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'qty': qty,
      'categoryId': categoryId,
      'urlImage': urlImage,
    };
  }
}
